File: catalog/backends/ruby-rails/patterns/inter_service_client.rb
# Pattern: Cross-service HTTP call — tracing headers propagated
# automatically by dd-trace-rb. Adapt: inject the target service
# URL via environment variable.
require "net/http"
require "json"

TARGET_URL = ENV.fetch("TARGET_SERVICE_URL", "http://localhost:8080")

def self.call(path)
  uri = URI("#{TARGET_URL}#{path}")
  response = Net::HTTP.get(uri)
  JSON.parse(response)
end

File: catalog/backends/node-express/patterns/inter_service_client.js
// Pattern: Cross-service HTTP call — tracing headers propagated
// automatically by dd-trace. Adapt: inject the target service
// URL via environment variable.
const TARGET_URL = process.env.TARGET_SERVICE_URL || 'http://localhost:8080';

async function call(path) {
  const resp = await fetch(`${TARGET_URL}${path}`);
  return resp.json();
}

module.exports = { call };
